pandas/io/json/json.py

```diff
@@ -263,6 +266,16 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
 
         .. versionadded:: 0.19.0
 
+    chunksize: integer, default None
+        If `lines=True`, how many lines to read into memory at a time.
+        If this is None, the file will be read into memory all at once.
+        Passing a chunksize helps with memory usage, but is slower.
+        Also note this is different from the `chunksize` parameter in
+            `read_csv`, which returns a FileTextReader.
```

Comment:
you need to return an iterator (as @jorisvandenbossche indicates).

you return a new class that inherits from ``BaseIterator`` (in pandas.io.common); see how ``pandas.io.stata.StateReader`` does this). You pretty much just return a class and define ``__next__``, which processes the chunked rows (also an example of this in pandas.io.pytables, though that doesn't use ``BaseIterator``, it should, which is another issue)